monitoring.go

```diff
@@ -123,6 +124,15 @@ os hostname <i>{{.Hostname}}</i><br>
 </html>
 `))
 
+func quitHandler(w http.ResponseWriter, r *http.Request) {
+	if r.Method != "POST" {
+		http.Error(w, "Use POST method for quitting", http.StatusMethodNotAllowed)
+		return
+	}
+
+	log.Fatalf("Received /quit")
```

Comment:
Using `log.Fatalf` instead of `os.Exit` means the client sending the request gets an empty reply. I'd personally go for a non fatal log and `os.Exit` (the goroutine+channel of my previous version isn't actually required).

Reply:
+1 your suggestion is much more reasonable.

Something like this?
```go
log.Infof("Received /quit")
http.Error(w, "OK quitting", http.StatusOK)
os.Exit(0)
```

I think the http sending is synchronous so there shouldn't be issues with the racing to exit, but tests will flush those out if they're lurking.

Do you want to amend this with your proposed change? If you're busy I'm also happy to do it myself, whatever you prefer.

Thanks a lot again!